src/components/backup/useCreateBackup.ts

```diff
@@ -49,7 +48,8 @@ export const useCreateBackup = ({ walletId, navigateToRoute }: UseCreateBackupPr
     [setLoading]
   );
   const onSuccess = useCallback(async () => {
-    if (!alreadyHasLocalPassword) {
+    const hasSavedPassword = await getLocalBackupPassword();
+    if (!hasSavedPassword) {
```

Comment:
@skylarbarrera also think the main issue here was the state variable wasn't be persisted when we were fetching the password on the BackupCloudStep component.

Reply:
yeah i think that makes sense to me